beacon-chain/core/blocks/signature.go

```diff
@@ -77,6 +78,31 @@ func VerifyBlockSignature(beaconState iface.ReadOnlyBeaconState,
 	return helpers.VerifyBlockSigningRoot(proposerPubKey, sig, domain, rootFunc)
 }
 
+// VerifyBlockSignature verifies the proposer signature of a beacon block. This differs
+// from the above method by not using fork data from the state and instead retrieving it
+// via the respective epoch.
+func VerifyBlockSignatureUsingSlot(beaconState iface.ReadOnlyBeaconState,
```

Comment:
Mainly due to the fact that block verification at the fork boundary fails with the older method as it uses the fork data from the state instead of the fork data as of the current epoch. The older method is used in a core method, so I don't want to deviate from the spec on this. I can refactor this to take the block instead.